examples/jsm/physics/RapierPhysics.js

```diff
@@ -1,220 +1,435 @@
-import { Clock, Vector3, Quaternion, Matrix4 } from 'three';
-
-const RAPIER_PATH = 'https://cdn.skypack.dev/@dimforge/rapier3d-compat@0.12.0';
-
-const frameRate = 60;
-
-const _scale = new Vector3( 1, 1, 1 );
-const ZERO = new Vector3();
-
+import {
+	Box3,
+	BufferGeometry,
+	IcosahedronGeometry,
+	InstancedMesh,
+	Matrix4,
+	Mesh,
+	Object3D,
+	Quaternion,
+	SphereGeometry,
+	Vector3,
+} from "three";
+
+/**
+ * @typedef {{} & import('@dimforge/rapier3d')} Rapier
+ */
+
+/**
+ * @typedef {{
+ *   rigidBodyDesc: Rapier.RigidBodyDesc;
+ *   rigidBody: Rapier.RigidBody;
+ *   colliderDesc: Rapier.ColliderDesc;
+ *   collider: Rapier.Collider;
+ * }} PhysicProperties
+ */
+
+/**
+ * @typedef {{
+ *   geometry?: BufferGeometry;
+ * } & Object3D } Object3DWithGeometry
+ */
+
+/**
+ * @initial_author mrdoob | info@mrdoob.com
+ *
+ * @description Physic helper based on `Rapier`
+ *
+ * @docs https://rapier.rs/docs/api/javascript/JavaScript3D/
+ */
+
+const RAPIER_PATH = "https://cdn.skypack.dev/@dimforge/rapier3d-compat@0.12.0";
+/** @type {Rapier | null | undefined} */
 let RAPIER = null;
 
-function getShape( geometry ) {
-
-	const parameters = geometry.parameters;
-
-	// TODO change type to is*
-
-	if ( geometry.type === 'BoxGeometry' ) {
-
-		const sx = parameters.width !== undefined ? parameters.width / 2 : 0.5;
-		const sy = parameters.height !== undefined ? parameters.height / 2 : 0.5;
-		const sz = parameters.depth !== undefined ? parameters.depth / 2 : 0.5;
-
-		return RAPIER.ColliderDesc.cuboid( sx, sy, sz );
-
-	} else if ( geometry.type === 'SphereGeometry' || geometry.type === 'IcosahedronGeometry' ) {
-
-		const radius = parameters.radius !== undefined ? parameters.radius : 1;
-		return RAPIER.ColliderDesc.ball( radius );
-
+export class Physic {
```

Comment:
I think it should be "Physics" instead of "Physic".

There are a few other instances of "Physic" in other variable names...